h/static/scripts/group-forms/hooks/use-group-annotations.tsx

```diff
@@ -42,6 +52,29 @@ export function useGroupAnnotations({
   const [totalAnnotations, setTotalAnnotations] = useState(0);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState<string>();
+  const updateAnnotationStatus = useCallback(
+    (annotationId: string, moderationStatus: ModerationStatus) => {
+      if (filterStatus === undefined) {
+        // If no filter status is set, simply change the annotation's status to
+        // the new one
+        setAnnotations(prev =>
+          prev?.reduce<APIAnnotationData[]>((annos, currentAnno) => {
+            if (currentAnno.id === annotationId) {
+              currentAnno.moderation_status = moderationStatus;
+            }
+
+            annos.push(currentAnno);
+            return annos;
+          }, []),
+        );
+      } else if (filterStatus !== moderationStatus) {
+        // If the moderation status is different from filter one, remove
+        // the annotation from the list
+        setAnnotations(prev => prev?.filter(anno => anno.id !== annotationId));
```

Comment:
This works since an annotation can't be re-added after being removed, except by changing the filter status which reloads annotations. An alternative approach would be to retain the local edits as separate state and compute the merged list from that. The current approach is fine at present.